base58.go

```diff
@@ -1,4 +1,4 @@
-package main
+package bc
```

Comment:
I'm afraid this is against the idea of the project: it should be a standalone app, not a package. Why did you decide to make this change?

Reply:
sure, this is up to you. I just add it for my purpose.

Reply:
hmm, actually, I am not comfortable with the GitHub pull request. 

Since what I want you to take is the fix on Decoder and merkle tree, the #2 - #4 commit.

While GitHub automatically create the pull request with all my new commits on master branch.

How could I send a pull request just with dedicated commit? or you could pick up the one you want? or I have to make a clean master branch so that you can pull?